### PR TITLE
Create Intel Module

### PR DESCRIPTION
**Pull request by bijx**
This feature adds a "Create Intel" module to the _Gameplay_ section which allows a Zeus to create custom intel objects for their missions on the fly. The Zeus can select a number of intel object types to make the vessel for their info:
image

The Zeus can then specify the title and description of the intel, whether the object should be deleted upon pickup, and how long the hold action duration should be:
image

Which creates the object that can be picked up:
image

Finally resulting in an object for the player and their selected companions:
image

### CODE REVIEW

File: Enhanced_Zeus_Modules.sqf [hunk 1]
```diff
@@ -11036,6 +11036,147 @@ MAZ_EZM_fnc_initFunction = {
...
+				"_this distance _target < 3",
+				"_caller distance _target < 3",  
+				{},
+				{},
+				{
+					private _intel = _this select 0; 
+					private _args = _this select 3;
+					_args params ["_title","_description","_deleteOnPickup", "_visibility"];
+
+					[_intel,0] remoteExec ["removeAction",0,true];
```
Comment (expung3d) [inline Enhanced_Zeus_Modules.sqf:11069]: Remove the `true` JIP parameter.

File: Enhanced_Zeus_Modules.sqf [hunk 2]
```diff
@@ -11036,6 +11036,147 @@ MAZ_EZM_fnc_initFunction = {
...
+					["IntelAdded",[format ["%1 picked up %2", (name _caller), _title], "a3\ui_f\data\igui\cfg\simpletasks\types\documents_ca.paa"]] remoteExec ['BIS_fnc_showNotification',0];
+
+					comment "for all players, remoteExec the createDiaryRecord";
+
+					private _intelParams = ["Diary", [format["%1", _title], format["%1", _description]]];
+
+					{
+						[_x, _intelParams] remoteExec ["createDiaryRecord", _x];
+					} forEach allPlayers;
```
Comment (expung3d) [inline Enhanced_Zeus_Modules.sqf:11078]: This creates a ton of `remoteExec` logs. Instead, use `allPlayers` as the target parameter for `remoteExec` so only one is needed.

File: Enhanced_Zeus_Modules.sqf (hunk 1 above)
Comment (expung3d) [reply Enhanced_Zeus_Modules.sqf:11069]: Also does this even remove the action? It's a hold action.

File: Enhanced_Zeus_Modules.sqf [hunk 3]
```diff
@@ -11036,6 +11036,147 @@ MAZ_EZM_fnc_initFunction = {
...
+					[
+						"EDIT",
+						"Title",
+						[
+								"",
+								1
+						]
+					],
+					[
+						"EDIT",
```
Comment (expung3d) [inline Enhanced_Zeus_Modules.sqf:11110]: Change to `"EDIT:MULTI"`

**Comment by bijx** [comment]
Tested on pub zeus server, player was able to pick up the intel and it shared with me as well.

File: Enhanced_Zeus_Modules.sqf [hunk 4]
```diff
@@ -11036,6 +11036,147 @@ MAZ_EZM_fnc_initFunction = {
...
+								0
+						]
+					],
+					[
+						"TOOLBOX",
+						"Delete on Pick Up",
+						[
+								true,
+								["No", "Yes"]
+						]
```
Comment (expung3d) [inline Enhanced_Zeus_Modules.sqf:11132]: Can change this to subtype `"TOOLBOX:YESNO"`. _Not necessary change._

File: Enhanced_Zeus_Modules.sqf [hunk 5]
```diff
@@ -11036,6 +11036,147 @@ MAZ_EZM_fnc_initFunction = {
...
+						"SLIDER",
+						"Hold Duration",
+						[
+							1,
+							12,
+							3,
+							objNull,
+							[1,1,1,0.7],
+							false
+						]
```
Comment (expung3d) [inline Enhanced_Zeus_Modules.sqf:11144]: The last 3 parameters for the slider aren't necessary, these are the defaults.

File: Enhanced_Zeus_Modules.sqf [hunk 6]
```diff
@@ -11036,6 +11036,147 @@ MAZ_EZM_fnc_initFunction = {
 
 	comment "Gameplay";
 
+		HYPER_EZM_fnc_handleCreateIntel = {
+			params ["_title","_description","_intelObject","_deleteOnPickup", "_target", "_holdDuration", "_visibility"];
+
+			comment "we set the default model as the documents for both redundancy and so we dont have to have a default case below";
```
Comment (expung3d) [inline Enhanced_Zeus_Modules.sqf:11042]: Why do this when you can provide a default in the `switch`?

```
private _intelObjModel = switch (_intelObject) do {
 case "laptop": {"Item_Laptop_Unfolded"};
 default {"Item_Laptop_Unfolded"};
};
```

File: Enhanced_Zeus_Modules.sqf [hunk 7]
```diff
@@ -11036,6 +11036,147 @@ MAZ_EZM_fnc_initFunction = {
...
+						params ["_values", "_args", "_display"];
+						systemChat "Create Intel dialog canceled.";
+						_display closeDisplay 2;
+				};
+				[
+						_dialogTitle,
+						_content,
+						_onConfirm,
+						_onCancel,
+						[_target]
```
Comment (expung3d) [inline Enhanced_Zeus_Modules.sqf:11174]: This doesn't need to be an array. You can pass the `_target` variable directly into `_args`.

File: Enhanced_Zeus_Modules.sqf [hunk 8]
```diff
@@ -11036,6 +11036,147 @@ MAZ_EZM_fnc_initFunction = {
...
+				private _onConfirm = {
+						params ["_values", "_args", "_display"];
+						_values params ["_title","_description","_intelObject","_deleteOnPickup","_holdDuration", "_visibility"];
+						private _target = _args # 0;
+						[ _title, _description, _intelObject, _deleteOnPickup, _target, _holdDuration, _visibility ] call HYPER_EZM_fnc_handleCreateIntel;
+						_display closeDisplay 1;
+				};
+				private _onCancel = {
+						params ["_values", "_args", "_display"];
+						systemChat "Create Intel dialog canceled.";
```
Comment (expung3d) [inline Enhanced_Zeus_Modules.sqf:11166]: I personally dislike `systemChats` when a dialog is closed. It's already evident to the user they closed the dialog, doesn't need to be stated again.

**Review by expung3d — CHANGES_REQUESTED** [review]
Just a handful of changes before it's good to go.

File: Enhanced_Zeus_Modules.sqf (hunk 1 above)
Comment (bijx) [reply Enhanced_Zeus_Modules.sqf:11069]: Yeah it should 
image

Comment (expung3d) [reply Enhanced_Zeus_Modules.sqf:11069]: Huh, didn't know that. Cool

File: Enhanced_Zeus_Modules.sqf (hunk 2 above)
Comment (bijx) [reply Enhanced_Zeus_Modules.sqf:11078]: https://community.bistudio.com/wiki/createDiaryRecord

I need to provide it with a "player" for some reason, so just doing `[_intelParams] remoteExec ["createDiaryRecord", allPlayers];` won't work. Is there a better way to reference the local player?

File: Enhanced_Zeus_Modules.sqf (hunk 8 above)
Comment (bijx) [reply Enhanced_Zeus_Modules.sqf:11166]: left that in by accident, it was for debugging 👍

File: Enhanced_Zeus_Modules.sqf (hunk 2 above)
Comment (expung3d) [reply Enhanced_Zeus_Modules.sqf:11078]: Why not just do a `spawn`?
```sqf
[_intelParams, {
    player createDiaryRecord _this;
}] remoteExec ["spawn",allPlayers];
```

**Review by expung3d — APPROVED** [review]

**Pull request merged**